azure/Guidelines.md

```diff
@@ -84,84 +82,114 @@ https://<tenant-id>-<service-defined-root>.<service>.azure.net
 
-#### Property is made required (from optional)
+There are a few mechanisms that can reduce breaking changes and their effects on our customers.
 
-If property “foo” was optional in the request body of v1 and now it is required, this should result in an API version change. If not changed, clients relying on the older API version will fail if this property is not passed.
+#### Use PATCH instead of PUT for updates
 
-#### URL format has changed
+The HTTP PUT verb is an idemopotent apply operation for the API version being used.  The [Microsoft API Guidelines already recommend the use of PATCH](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#742-patch) for updates.
 
-Resource parameter names change from `/resourceType1/{resourceType1_name}` to `/resourceType1/{resourceType1_id}`. This will impact code generation.
+Consider the following sequence:
 
-#### Resource naming rules should not change
+* User1 creates a resource with version v2, using a new optional parameter.
+* Later, User2 wants to update the resource using unrelated settings.  Using version v1, User2 issues a GET, does the changes, and then issues a PUT to replace the resource definition.
 
-This could result in failures which would have earlier succeeded. Even if the rules become less strict, clients relying on earlier name constraints to perform local validation will fail.
+In this case, the optional parameter is lost because of the replace semantics.  The optional parameter only exists on API version v2, and not on version v1.
 
-### Non-Breaking Changes
+Service teams SHOULD prefer and recommend PATCH operations for updating resources.
 
-The following changes are considered backwards compatible and hence non-breaking.
+#### Use extensible enums
 
-#### Adding new APIs to an existing service
+While removing a value from an enum is a breaking change, adding an enum can be handled with an _extensible enum_.  An extensible enum is a string value that has been marked with a special marker - setting `modelAsString` to true within an `x-ms-enum` block.  For example:
 
-When a new resource types is added, it does not require API version to be updated for existing types.
-
-#### Bug fixes to existing API
+```json
+"createdByType": {
+   "type": "string",
+   "description": "The type of identity that created the resource.",
+   "enum": [
+      "User",
+      "Application",
+      "ManagedIdentity",
+      "Key"
+   ],
+   "x-ms-enum": {
+      "name": "createdByType",
+      "modelAsString": true
+   }
+}
+```
 
-Bug fixes to existing API which don’t fall into one of the above categories of breaking changes as described above are fine.
+If you can foresee that an enum will be extended in the future, model the enum as an extensible enum.
```

Comment:
I"d rephrase this: If you are positive that the enum's symbol set will NEVER change over time, then use model as an enum otherwise, model the enum as a string.